x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -511,27 +543,14 @@ private static boolean verifyBcryptHash(SecureString text, char[] hash) {
 
     public abstract boolean verify(SecureString data, char[] hash);
 
-
-    static final class SaltProvider {
-
-        private SaltProvider() {
-            throw new IllegalStateException("Utility class should not be instantiated");
-        }
-
-        static final char[] ALPHABET = new char[]{
-            '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D',
-            'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T',
-            'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
-            'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'
-        };
-
-        public static char[] salt(int length) {
-            Random random = Randomness.get();
-            char[] salt = new char[length];
-            for (int i = 0; i < length; i++) {
-                salt[i] = ALPHABET[(random.nextInt(ALPHABET.length))];
-            }
-            return salt;
-        }
+    /**
+     * Generates an array of {@code length} random bytes using {@link java.security.SecureRandom}
+     */
+    private static byte[] generateSalt(int length) {
+        Random random = new SecureRandom();
```

Comment:
can we store the value? SecureRandom objects can be reused and we don't need to keep creating them